src/main/java/org/dataone/hashstore/filehashstore/FileHashStore.java

```diff
@@ -554,40 +550,32 @@ public void tagObject(String pid, String cid) throws IOException, PidRefsFileExi
         NoSuchAlgorithmException, FileNotFoundException, InterruptedException {
         logFileHashStore.debug("Tagging cid (" + cid + ") with pid: " + pid);
         // Validate input parameters
-        FileHashStoreUtility.ensureNotNull(pid, "pid", "tagObject");
-        FileHashStoreUtility.ensureNotNull(cid, "cid", "tagObject");
-        FileHashStoreUtility.checkForNotEmptyAndValidString(pid, "pid", "tagObject");
-        FileHashStoreUtility.checkForNotEmptyAndValidString(cid, "cid", "tagObject");
+        FileHashStoreUtility.ensureNotNull(pid, "pid");
+        FileHashStoreUtility.ensureNotNull(cid, "cid");
+        FileHashStoreUtility.checkForNotEmptyAndValidString(pid, "pid");
+        FileHashStoreUtility.checkForNotEmptyAndValidString(cid, "cid");
 
         try {
-            synchronizeObjectLockedCids(cid);
-            synchronizeReferenceLockedPids(pid);
             storeHashStoreRefsFiles(pid, cid);
 
         } catch (HashStoreRefsAlreadyExistException hsrfae) {
+            // *** cid and pid already released ***
```

Comment:
remove unless you really like it (I just included it in the example to provide a bit of explanation. It seems superfluous in the actual code)